### PR TITLE
remove #include (doxygen)

### PR DESCRIPTION
**Pull request by pavel-pimenov**
before:
include-graph-before

after:
include-graph-after

### CODE REVIEW

File: src/session.cpp [hunk 1]
```diff
@@ -34,8 +34,6 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/extensions/ut_pex.hpp"
 #include "libtorrent/extensions/ut_metadata.hpp"
 #include "libtorrent/extensions/smart_ban.hpp"
-#include "libtorrent/session.hpp"
```
Comment (aldenml) [inline src/session.cpp]: just a comment: I'm in favor of cleanup as much as possible, but in my opinion, the `.cpp` should include the corresponding `.hpp` if only for coding style.

Comment (arvidn) [inline src/session.cpp]: I had the same thought.

File: src/session_handle.cpp [hunk 2]
```diff
@@ -30,8 +30,6 @@ POSSIBILITY OF SUCH DAMAGE.
 
 */
 
-#include "libtorrent/session_handle.hpp"
```
Comment (aldenml) [inline src/session_handle.cpp]: same here

File: include/libtorrent/aux_/typed_span.hpp [hunk 3]
```diff
@@ -33,11 +33,8 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_TYPED_SPAN_HPP
 #define TORRENT_TYPED_SPAN_HPP
 
-#include <type_traits>
```
Comment (arvidn) [inline include/libtorrent/aux_/typed_span.hpp]: this header is required for ``std::enable_if``

File: include/libtorrent/aux_/typed_span.hpp [hunk 4]
```diff
@@ -33,11 +33,8 @@ POSSIBILITY OF SUCH DAMAGE.
...
-
 #include "libtorrent/units.hpp"
 #include "libtorrent/span.hpp"
-#include "libtorrent/assert.hpp"
```
Comment (arvidn) [inline include/libtorrent/aux_/typed_span.hpp]: this header is required for ``TORRENT_ASSERT``

File: include/libtorrent/aux_/ffs.hpp [hunk 5]
```diff
@@ -34,7 +34,6 @@ POSSIBILITY OF SUCH DAMAGE.
 #define TORRENT_FFS_HPP_INCLUDE
 
 #include <cstdint>
-#include "libtorrent/export.hpp"
```
Comment (arvidn) [inline include/libtorrent/aux_/ffs.hpp]: this header is required for ``TORRENT_EXTRA_EXPORT``

File: include/libtorrent/bitfield.hpp [hunk 6]
```diff
@@ -40,7 +40,6 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/aux_/ffs.hpp"
 
 #include <cstring> // for memset and memcpy
-#include <cstdint> // uint32_t
```
Comment (arvidn) [inline include/libtorrent/bitfield.hpp]: this header is required for ``std::uint32_t``

File: include/libtorrent/config.hpp [hunk 7]
```diff
@@ -37,7 +37,6 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #define _FILE_OFFSET_BITS 64
 
-#include <boost/config.hpp>
```
Comment (arvidn) [inline include/libtorrent/config.hpp]: I believe this is required for ``BOOST_MSVC``

File: include/libtorrent/peer_info.hpp [hunk 8]
```diff
@@ -39,7 +39,6 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/config.hpp"
 #include "libtorrent/bitfield.hpp"
 #include "libtorrent/time.hpp"
-#include "libtorrent/units.hpp"
```
Comment (arvidn) [inline include/libtorrent/peer_info.hpp]: I believe this is required for ``piece_index_t``

File: include/libtorrent/sha1_hash.hpp [hunk 9]
```diff
@@ -35,10 +35,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include <cctype>
 #include <algorithm>
-#include <string>
```
Comment (arvidn) [inline include/libtorrent/sha1_hash.hpp]: this is required for ``std::string``

File: include/libtorrent/sha1_hash.hpp [hunk 10]
```diff
@@ -35,10 +35,8 @@ POSSIBILITY OF SUCH DAMAGE.
...
 #include <cstring>
 
-#include "libtorrent/config.hpp"
```
Comment (arvidn) [inline include/libtorrent/sha1_hash.hpp]: this is required for ``TORRENT_USE_IOSTREAM`` and ``TORRENT_EXTRA_EXPORT`` and probably a bunch of other macros

**Comment by arvidn** [comment]
I didn't go through all of the changes, but it seems this breaks header independency. headers should be self contained, as in, they should not yield any errors if they are the first to be included in a translation unit.

File: include/libtorrent/aux_/typed_span.hpp (hunk 3 above)
Comment (pavel-pimenov) [inline include/libtorrent/aux_/typed_span.hpp]: libtorrent/span.hpp
...
#include <type_traits> 

default

File: include/libtorrent/aux_/typed_span.hpp (hunk 4 above)
Comment (pavel-pimenov) [inline include/libtorrent/aux_/typed_span.hpp]: span.hpp :
...
#include "libtorrent/assert.hpp"

File: include/libtorrent/aux_/ffs.hpp (hunk 5 above)
Comment (pavel-pimenov) [inline include/libtorrent/aux_/ffs.hpp]: default

File: include/libtorrent/bitfield.hpp (hunk 6 above)
Comment (pavel-pimenov) [inline include/libtorrent/bitfield.hpp]: default

File: include/libtorrent/config.hpp (hunk 7 above)
Comment (pavel-pimenov) [inline include/libtorrent/config.hpp]: default

File: include/libtorrent/peer_info.hpp (hunk 8 above)
Comment (pavel-pimenov) [inline include/libtorrent/peer_info.hpp]: default

File: include/libtorrent/sha1_hash.hpp (hunk 10 above)
Comment (pavel-pimenov) [inline include/libtorrent/sha1_hash.hpp]: default

File: include/libtorrent/aux_/ffs.hpp (hunk 5 above)
Comment (arvidn) [inline include/libtorrent/aux_/ffs.hpp]: sure, there are transitive dependencies on that header elsewhere, that doesn't make it a good idea to depend on them. This is what include guards are for.

**Pull request closed without merging**